src/rust/generators/generic-subgraph-generator/src/generator.rs

```diff
@@ -60,10 +74,10 @@ where
     pub(crate) async fn convert_events_to_subgraph(
         &mut self,
         events: Vec<GenericEvent>,
-    ) -> (Graph, Vec<impl Cacheable>, Option<eyre::Report>) {
+        completed: &mut CompletedEvents,
+    ) -> Result<Graph, Result<(Graph, GenericSubgraphGeneratorError), GenericSubgraphGeneratorError>> {
```

Comment:
personally i actually liked the total / partial / whatever enum, but to each their own. it's less immediately obvious what an Err(Ok(tuple)) means